### DISCUSSION TITLE
How to hide a window from a new thread?

### DISCUSSION THREAD

**Discussion by felixmaker** [fltk-rs/fltk-rs#1442] [Q&A] [answered]
I use `somewindow.hide()` method in a new thread. However, it panicked with: 

```
thread '<unnamed>' panicked at 'assertion failed: crate::app::is_ui_thread()',
```

How to hide a window from a new thread?

**Comment by MoAlyousef** [comment, marked as answer]
The FLTK documentation describes it as unsafe to hide a window from a different thread other than the ui thread.
You could either use a channel, send a message that you intercept in the main ui thread to hide the window.
Otherwise you could use:
```rust
unsafe { fltk_sys::window::Fl_Window_hide(wind.as_widget_ptr() as _); }
```

**Reply by felixmaker** [reply]
Thank you!
